nuitka/build/Backend.scons

```diff
@@ -313,6 +313,10 @@ mingw_mode = prepareEnvironment(mingw_mode=mingw_mode)
 # Patch the compiler detection.
 Environment.Detect = getEnhancedToolDetect()
 
+# Because after call `createEnvironment` os.environ will change.
```

Comment:
I see, good catch, indeed if that's overwritten, but then assumed as user provided, which cannot be good. I thought we do the overwrite in env only later.